src/spootnik/reporter/sentry.clj

```diff
@@ -0,0 +1,98 @@
+(ns spootnik.reporter.sentry
+  (:require
+   [sentry-clj.core :as sentry-io]
+   [manifold.deferred  :as d]
+   [clojure.string :as str]
+   [clojure.tools.logging :refer [error]]
+   [clojure.java.shell :as sh]))
+
+(def http-requests-payload-stub
+  "Storage for stubbed http sentry events "
+  (atom nil))
+
+(defn in-memory? [dsn]
+  (or (= dsn ":memory:")
+      (nil? dsn)))
+
+(def hostname-refresh-interval
+  "How often to allow reading /etc/hostname, in seconds."
+  60)
+
+(defn get-hostname
+  "Get the current hostname by shelling out to 'hostname'"
+  []
+  (or
+   (try
+     (let [{:keys [exit out]} (sh/sh "hostname")]
+       (when (= exit 0)
+         (str/trim out)))
+     (catch Exception _))
+   "<unknown>"))
+
+(defn hostname
+  "Fetches the hostname by shelling to 'hostname', whenever the given age
+  is stale enough. If the given age is recent, as defined by
+  hostname-refresh-interval, returns age and val instead."
+  [[age val]]
+  (if (and val (<= (* 1000 hostname-refresh-interval)
+                   (- (System/currentTimeMillis) age)))
+    [age val]
+    [(System/currentTimeMillis) (get-hostname)]))
+
+(let [cache (atom [nil nil])]
+  (defn localhost
+    "Returns the local host name."
+    []
+    (if (re-find #"^Windows" (System/getProperty "os.name"))
+      (or (System/getenv "COMPUTERNAME") "localhost")
+      (or (System/getenv "HOSTNAME")
+          (second (swap! cache hostname))))))
+
+(defn- e->sentry-request [e]
+  (let [payload (-> e :extra :payload)]
+    {:url          (-> payload :uri)
+     :method       (-> payload :request-method name clojure.string/upper-case)
+     :query-string (-> payload :query-string)
+     :headers      (-> payload :headers)}))
+
+(defn e->sentry-event [e options tags]
+  (let [{:keys [message extra throwable]} e]
+
+    (if message
```

Comment:
Do we really want an `if`? Some callers might want to supply both `message` and `tags`, etc
Why not just pass all the fields that were received? (eg via `cond-> ... `)

Reply:
message AND tags is a valid point..
..
coz in some cases we receive Exception that needs to be parsed, but in some cases - just regular map with a msg key..
?